### ISSUE TITLE
Stop forcing -march=native

### ISSUE THREAD

**Issue by jengelh** [libsemigroups/libsemigroups#239]
Please honor the user-provided CFLAGS and do not override them. If anything, put own flags in the same place as -O3 (O3's location is good).

```
./configure CFLAGS=-march=znver1 CXXFLAGS=-march=znver1 && make
[...]
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I./config  -DNDEBUG -DHPCOMBI_CONSTEXPR_FUN_ARGS     -I~/libsemigroups/include -I~/libsemigroups/extern -std=gnu++11 -O3 -Wall -Wextra   -I./extern/eigen-3.3.7/   -march=znver1 -march=native -flax-vector-conversions -MT src/libsemigroups_la-blocks.lo -MD -MP -MF src/.deps/libsemigroups_la-blocks.Tpo -c -o src/libsemigroups_la-blocks.lo `test -f 'src/blocks.cpp' || echo './'`src/blocks.cpp
```

**Comment by james-d-mitchell** [comment]
Thanks for the report @jengelh. Have you tried compiling with `./configure --disable-hpcombi`, this should mean that the offending flags never get added in the first place. I'm not sure that `libsemigroups` will compile with HPCombi support enabled but without the flag `-march=native` being set. I have a proper fix to this too in case it is possible to compile with HPCombi support enabled and `-march=znver1`

**Comment by jengelh** [comment]
>I'm not sure that libsemigroups will compile with HPCombi support enabled but without the flag -march=native being set.

Why not? native would set znver2 for me, which is not overly different from znver1.

**Comment by jengelh** [comment]
>I'm not sure that libsemigroups will compile with HPCombi support enabled but without the flag -march=native being set

Why not—what would make {using `-march=znver1` on a znver2 machine} different from building {using `-march=native` on a znver1 machine}?

**Comment by james-d-mitchell** [comment]
I have no idea what a znver2 machine is, HPCombi makes use of the AVX instruction set, if that's supported on znver2 machines, then perhaps it'll work.

**Comment by jengelh** [comment]
znver1 is Zen 1, aka Ryzen 1600, 1700, 2600, etc.
znver2 is Zen 2, aka Ryzen 3700, 3850, etc.
All of them have AVX1 and AVX2.

**Comment by james-d-mitchell** [comment]
Ok, I'll push my proper fix that appends the `-march=native` and `-flax-vector-conversions` in the same place as the `-O3`, that'll resolve this, right?

**Comment by jengelh** [comment]
sure.

**Comment by james-d-mitchell** [comment]
Resolved in v1.3.2
